create.js

```diff
@@ -23,10 +23,25 @@ module.exports = function(_nativeProtocols) {
   return publicApi;
 
   function execute(options) {
-    var clientRequest;
     var fetchedUrls = [];
-
-    return (clientRequest = cb());
+    var clientRequest = cb();
+    var requestProxy = Object.create(clientRequest);
+
+    // separate the proxy's event handlers from the request's
+    requestProxy._events = {};
+    requestProxy._eventsCount = 0;
+    if (options.userCallback) {
+      requestProxy.on('response', options.userCallback);
+    }
+    // forward all request events on the proxy except `response`
```

Comment:
An alternative would be to forward _no_ events, except `response` and `error` (already performed with `forwardError` below).